integration/deltaspike-data/impl/src/main/java/com/blazebit/persistence/impl/deltaspikedata/impl/meta/extractor/EntityViewAnnotationMetadataExtractor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2014 - 2017 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.impl.deltaspikedata.impl.meta.extractor;
+
+import com.blazebit.persistence.impl.deltaspikedata.impl.meta.RepositoryEntityView;
+import com.blazebit.persistence.view.EntityView;
+import org.apache.deltaspike.data.api.Repository;
+import org.apache.deltaspike.data.impl.util.EntityUtils;
+
+/**
+ * Implementation is similar to {@link org.apache.deltaspike.data.impl.meta.extractor.AnnotationMetadataExtractor} but was modified to
+ * work with entity views.
+ *
+ * @author Moritz Becker
+ * @since 1.2.0
+ */
+public class EntityViewAnnotationMetadataExtractor implements EntityViewMetadataExtractor {
+
+    @Override
+    public RepositoryEntityView extract(Class<?> repoClass) {
+        Repository repo = repoClass.getAnnotation(Repository.class);
+        Class<?> repoEntity = repo.forEntity();
+        EntityView entityViewAnnotation = repoEntity.getAnnotation(EntityView.class);
```

Comment:
Would be great if it was possible to query the `EntityViewManager` for this check instead.

Reply:
Not possible since this happens within ProcessAnnotatedType in the CDI extension and the EntityViewManager is not available at that time.